Telink_TC32/data/languages/Telink_TC32.slaspec

```diff
@@ -7,19 +7,19 @@
 # Memory architecture
 define endian=little;
 define alignment=2;  # TODO: Verify. See 4.2 "Alignment Definition."
-define space RAM type=ram_space size=2 default;
-define space register type=register_space size=2;
+define space RAM type=ram_space size=4 default;
+define space register type=register_space size=4;
 
 
 # Registers
-define register offset=0x0000 size=2 [
+define register offset=0x0000 size=4 [
   r0 r1 r2 r3 r4 r5 r6 r7 r8 r9 r10 r11 r12 sp lr pc
```

Comment:
Could you tell me where you found that these registers are supposed to be 32-bit?

Reply:
Registers are definitely 32bit wide because they are used to address 128KB / 512KB of flash memory. There is the same function structure like ARM where addresses are stored after return instruction and are referenced like " tloadr	r3, [pc, #12]" which takes address 0x0080a538 in my sample. Also there are different opcodes for loadr / loadrb / loadrh for the different sizes.